### PR TITLE
Option to skip SSL verification with LDAP auth backend

### PR DESCRIPTION
**Pull request by esell**
Adding a new boolean field to allow users to skip verification of the LDAP servers SSL certificate when using the LDAP auth backend.

### CODE REVIEW

File: builtin/credential/ldap/path_config.go [hunk 1]
```diff
@@ -100,6 +106,10 @@ func (b *backend) pathConfigWrite(
 	if groupdn != "" {
 		cfg.GroupDN = groupdn
 	}
+	sslverify := d.Get("sslverify").(bool)
+	if !sslverify {
+		cfg.SSLVerify = sslverify
```
Comment (sethvargo) [inline builtin/credential/ldap/path_config.go]: I'm a big fan of adding a very large warning message here explaining that disabling SSL verification is bad and is not secure.

Comment (esell) [inline builtin/credential/ldap/path_config.go]: Would a warning in the help output be enough? I'm looking in the code now and I don't see a great spot to put an output on the actual write. In testing I changed pathConfigWrite() to send back a logical.Response even on success but that doesn't appear to be super clean as the actual "Success!" write message then gets trampled on.
I did update the code to make ignoring SSL verification non-default.

File: builtin/credential/ldap/path_config.go [hunk 2]
```diff
@@ -31,6 +32,10 @@ func pathConfig(b *backend) *framework.Path {
 				Type:        framework.TypeString,
 				Description: "Attribute used for users (default: cn)",
 			},
+			"skipsslverify": &framework.FieldSchema{
```
Comment (armon) [inline builtin/credential/ldap/path_config.go]: Could you please change this to`tls_skip_verify`?

**Comment by armon** [comment]
After the rename, this requires changes to the documentation to stay in sync

**Comment by esell** [comment]
"skipsslverify" has been changed to "insecure_tls" and any references have been updated accordingly.

**Comment by armon** [comment]
Thanks!

**Pull request merged**